### PR TITLE
New button variants and ranks

### PR DESCRIPTION
**Pull request by janeszelag**
[Word Document on new variants](https://docs.google.com/document/d/19sZaR43PT6poWr0ykjgcVio_ECdVaB7TrlNCZ78O-uk/edit#)
[Jira Ticket](https://telusdigital.atlassian.net/browse/DJR-2893)
[Designs in Invision](https://telus.invisionapp.com/d/main#/console/13491097/432314908/preview)

- [x] Desk check was done with @Christina-Lo for functionality and documentation

Adds the following new Button variants of TDS core buttons along with usage guidelines/criteria:
- Standard: Main/Common
- Brand: Main/Common
- Danger: Common

Oct-15-2020 11-51-50
Screen Shot 2020-10-15 at 11 50 44 AM
Screen Shot 2020-10-15 at 11 50 56 AM

### CODE REVIEW

File: packages/Button/Button.md [hunk 1]
```diff
@@ -5,13 +5,17 @@ Provide a function as the `onClick` prop to perform an action when clicked. **Av
...
+- Keep the text short and precise. It should fit on a single line (our recommendation is 2-24 characters for French and English)
+- Make use of the A11yContent core component to provide more written context for assistive technology users for when they navigate a page using only button landmarks
 - Make sure the button text describes an action
 - Buttons should not be disabled. Use inline (on blur) error messaging to provide feedback when the form is invalid
 
-By default, Buttons will be displayed in the `primary` variant. Use primary buttons for the main action on a page or
-in a form.
+### OLD Button variants
+
+**NOTE:** New Button variants have been introduced. Future projects should adopt the new variants, as the older variants will have a deprecation plan. Avoid using both new and old variants together within the same experience.
```
Comment (jraff) [inline packages/Button/Button.md:16]: Can we change this to:

**Deprecated:** New Button variants have been introduced. Future projects should adopt the new variants, as the older variants will be deprecated. Avoid using both new and old variants together within the same experience.

**Review by jraff — CHANGES_REQUESTED** [review]
Can we update the `CartesianButton` with the new variants and ranks?

File: packages/Button/Button.jsx [hunk 2]
```diff
@@ -18,43 +26,131 @@ const preventDisabling = ({ disabled, ...props }) => {
...
+      }
+      break
+
+    case 'danger':
+      backgroundColor = colorWhite
+      color = colorCardinal
+      border = `0.0625rem solid ${colorCardinal}`
+      hover.boxShadow = `0 0 0 0.125rem ${colorCardinal}`
+      active.color = '#770F1B'
+      active.backgroundColor = '#FFF6F8 !important'
```
[the remark this answers is not in the document]
Comment (janeszelag) [reply packages/Button/Button.jsx:120]: done:)

File: packages/Button/Button.jsx [hunk 3]
```diff
@@ -18,43 +26,131 @@ const preventDisabling = ({ disabled, ...props }) => {
...
+        active.backgroundColor = '#231332 !important'
+        focus.backgroundColor = '#371E4F'
+        focus.boxShadow = `0 0 0 0.1875rem #7C53A5 , 0 0 0 0.125rem ${colorWhite} inset`
+      } else {
+        backgroundColor = colorWhite
+        color = colorTelusPurple
+        border = `0.0625rem solid ${colorTelusPurple}`
+        hover.boxShadow = `0 0 0 0.125rem ${colorTelusPurple}`
+        active.color = '#371E4F'
+        active.backgroundColor = '#F2EFF4 !important'
```
[the remark this answers is not in the document]
Comment (janeszelag) [reply packages/Button/Button.jsx:108]: done:)

File: packages/Button/Button.md (hunk 1 above)
Comment (janeszelag) [reply packages/Button/Button.md:16]: done :)

File: packages/Button/Button.jsx [hunk 4]
```diff
@@ -94,6 +199,7 @@ Button.propTypes = {
 Button.defaultProps = {
   type: 'button',
   variant: 'primary',
+  rank: 'common',
```
Comment (canrozanes) [inline packages/Button/Button.jsx:202]: as per [this doc](https://docs.google.com/document/d/19sZaR43PT6poWr0ykjgcVio_ECdVaB7TrlNCZ78O-uk/edit#), buttons by default are supposed to be of type `rank="main"`, why are we passing `rank="common"` as default prop?

File: packages/Button/Button.md [hunk 5]
```diff
@@ -23,10 +27,61 @@ Specify the `variant` to create a button for secondary actions.
...
+
+```jsx
+<Button variant="standard">Standard Common</Button>
+```
+
+```jsx
+<Button variant="standard" rank="main">
+  Standard Main
+</Button>
+```
```
Comment (canrozanes) [inline packages/Button/Button.md:54]: if main is indeed the default, shouldn't it come first in documentation?

**Review by canrozanes — CHANGES_REQUESTED** [review]

File: packages/Button/Button.jsx (hunk 4 above)
Comment (janeszelag) [reply packages/Button/Button.jsx:202]: default is common, discussion on slack to confirm

File: packages/Button/Button.md (hunk 5 above)
Comment (janeszelag) [reply packages/Button/Button.md:54]: it is not.

**Review by jraff — APPROVED** [review]

**Pull request merged**